.github/workflows/build.yml

```diff
@@ -37,6 +37,7 @@ jobs:
           ${{format(matrix.generator != 'Default Generator' && '-G "{0}"' || '', matrix.generator)}}
         CMAKE_INSTALL_PREFIX: "${{ github.workspace }}/install-prefix"
         CMAKE_BUILD_TYPE: Debug
+        CMAKE_CXX_FLAGS_DEBUG: ${{ matrix.googletest == 'build' && '-g -D_GLIBCXX_DEBUG -D_GLIBCXX_DEBUG_PEDANTIC' || '-g' }} 
```

Comment:
Why did you put the googletest == build clause here? Does the system googletest fail with these flags?

Reply:
Yes, 'system' failed to build, just 'build' worked.

I didn't dig deeper, but it's a common effect in the _debug_ mode of the GNU standard C++ library that it can **only**  be mixed with non-_debug_ binaries when those do **not** exchange data with the _debug_-generated code in e.g. ˋstd::ˋ containers or smart pointers. I thus assumed that the system-provided _googletest_ (which is obviously non-_debug_) doesn't mix with the _debug_ structures, by design. So I excluded these variants from being built in the _debug_ mode.

Reply:
OK, sounds good. Can you add a comment explaining this?